### DISCUSSION TITLE
conditional "multiple" prop in ComboBox

### DISCUSSION THREAD

**Discussion by mrxsal** [tailwindlabs/headlessui#2604] [Help]
I've come across a Typescript  issue which I don't fully understand.
I am building a custom component based on headlessui Combobox. I'd like to pass the `multiple` prop dynamically to the ComboBox element:

```
interface MyComponentProps {
  multiple?: boolean;
}
```

in MyComponent  > 

`<Combobox
        multiple={multiple}
      >`

I am getting the following TS error:
```
Type 'boolean' is not assignable to type 'true'.
The expected type comes from property 'multiple' which is declared here on type 'IntrinsicAttributes & ComboboxProps<{ label: string | number; value: string | number; } | Option[] | null, false, true, ExoticComponent<{ children?: ReactNode; }>>'
```

I can set `multiple={true}` or `multiple={false}` statically, but a boolean prop doesn't work? 

Is there a reason why multiple has to be `true` or not passed at all?

**Comment by bendadaniel** [comment]
Same problem with nullable

**Comment by ValentinaPanic** [comment]
I am having the same issue as @mrxsal. Did anyone find a good solution for this Typescript issue?

**Comment by alanPaivaaspOctos** [comment]
https://github.com/tailwindlabs/headlessui/discussions/2821
I've opened this discution to see if it's possible to allow such thing

**Comment by andredewaard** [comment]
This is actually possible on the listbox, i don't get why this shouldn't be working on the combobox, i have the same [issue](https://github.com/tailwindlabs/headlessui/discussions/2873).
